pkg/client/unversioned/transport.go

```diff
@@ -171,30 +171,43 @@ func NewClientCertTLSConfig(certData, keyData, caData []byte) (*tls.Config, erro
 	if err != nil {
 		return nil, err
 	}
-	certPool := x509.NewCertPool()
-	certPool.AppendCertsFromPEM(caData)
+
+	caPool := getCertPoolForCA(caData)
+
 	return &tls.Config{
 		// Change default from SSLv3 to TLSv1.0 (because of POODLE vulnerability)
 		MinVersion: tls.VersionTLS10,
 		Certificates: []tls.Certificate{
 			cert,
 		},
-		RootCAs:    certPool,
-		ClientCAs:  certPool,
+		RootCAs:    caPool,
+		ClientCAs:  caPool,
```

Comment:
setting ClientCAs/ClientAuth doesn't make sense here... those are settings used by the server, not the client